pkg/apis/datasource/v0alpha1/config.go

```diff
@@ -0,0 +1,76 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type GenericDataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// generic config
+	Spec GenericDataSourceSpec `json:"spec"`
+
+	// Secure values placeholder (true for fields that exist)
+	Secure map[string]SecureValue `json:"secure,omitempty"`
+}
+
+type SecureValue struct {
+	// The input is only valid for writing the value -- it is replaced on read
+	Input string `json:"input,omitempty"`
+
+	// The name identifier for this secure value
+	Reference string `json:"ref,omitempty"`
```

Comment:
Just to clarify: is this the secure value name of an already existing secure value? Meaning you can either set the `Input` or `Reference` (an union)

Reply:
Yes in the current form, the idea is you could either write a raw value or a refeence.

When you write a raw value, the server would create a secret for you and replace it with a reference